integration/spark/src/main/common/java/io/openlineage/spark/agent/facets/builder/EnvironmentFacetBuilder.java

```diff
@@ -0,0 +1,101 @@
+package io.openlineage.spark.agent.facets.builder;
+
+import io.openlineage.spark.agent.facets.EnvironmentFacet;
+import io.openlineage.spark.api.CustomFacetBuilder;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.*;
+import java.util.function.BiConsumer;
+
+import io.openlineage.spark.api.OpenLineageContext;
+import org.apache.spark.scheduler.SparkListenerEvent;
+import org.json.simple.parser.JSONParser;
+
+public class EnvironmentFacetBuilder extends CustomFacetBuilder<SparkListenerEvent, EnvironmentFacet> {
+    private static HashMap<String, Object> dbProperties;
+    private static final org.slf4j.Logger log
+            = org.slf4j.LoggerFactory.getLogger(EnvironmentFacetBuilder.class);
+    private final OpenLineageContext openLineageContext;
+
+    public EnvironmentFacetBuilder(OpenLineageContext openLineageContext) {
+        this.openLineageContext = openLineageContext;
+    }
+
+    @Override
+    protected void build(SparkListenerEvent event, BiConsumer<String, ? super EnvironmentFacet> consumer) {
+        if(System.getenv().containsKey("DATABRICKS_RUNTIME_VERSION")) {
+            consumer.accept("environment-properties", new EnvironmentFacet(getDatabricksEnvironmentalAttributes(event)));
+        }
+    }
+
+    private HashMap<String, Object> getDatabricksEnvironmentalAttributes(SparkListenerEvent jobStart) {
+        dbProperties = new HashMap<>();
+        String urlString = "http://" + openLineageContext.getProperties().getProperty("spark.databricks.clusterUsageTags.driverInstancePrivateIp") + ":7070/?type=%22com.databricks.backend.daemon.data.common.DataMessages$GetMountsV2%22";
+
+        List<String> dbPropertiesKeys = Arrays.asList(
+                "orgId",
+                "spark.databricks.clusterUsageTags.clusterOwnerOrgId",
+                "spark.databricks.notebook.path",
+                "spark.databricks.job.type",
+                "spark.databricks.job.id",
+                "spark.databricks.job.runId",
+                "user",
+                "userId",
+                "spark.databricks.clusterUsageTags.clusterName",
+                "spark.databricks.clusterUsageTags.azureSubscriptionId"
+        );
+
+        dbPropertiesKeys.stream().forEach((p) -> {
+            dbProperties.put(p, openLineageContext.getProperties().getProperty(p));
+        });
+
+        dbProperties.put("mountPoints", getDatabricksMountpoints(urlString));
+
+        return dbProperties;
+    }
+
+    private static List<Map<String, String>> getDatabricksMountpoints(String urlString) {
+        List<Map<String, String>> mapList = new ArrayList<>();
+        try {
+            URL url = new URL(urlString);
+            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
```

Comment:
You can use Apache HTTP client - it's one of the few dependencies we include directly.